File: cpp/difference-of-squares/difference_of_squares.cpp
#include "difference_of_squares.h"

int squares::square_of_sums(const int num) {
  int sum = 0;
  for (int i = 1; i <= num; i++) {
    sum += i;
  }
  return sum * sum;
}

int squares::sum_of_squares(const int num) {
  int sum = 0;
  for (int i = 1; i <= num; i++) {
    sum += i * i;
  }
  return sum;
}

int squares::difference(const int num) {
  return square_of_sums(num) - sum_of_squares(num);
}

File: javascript/difference-of-squares/difference-of-squares.js
export class Squares {
  constructor(num) {
    this._numbers = [...Array(num)].map((_, i) => i + 1);
  }

  get sumOfSquares() {
    return this._numbers.reduce((sum, n) => sum += n ** 2, 0);
  }

  get squareOfSum() {
    return this._numbers.reduce((sum, n) => sum += n, 0) ** 2
  }

  get difference() {
    return this.squareOfSum - this.sumOfSquares;
  }
}
